libraries/c_sdk/standard/mqtt/src/iot_mqtt_operation.c

```diff
@@ -849,6 +849,9 @@ void _IotMqtt_ProcessKeepAlive( IotTaskPool_t pTaskPool,
     {
         _IotMqtt_CloseNetworkConnection( IOT_MQTT_KEEP_ALIVE_TIMEOUT,
                                          pMqttConnection );
+
+        /* Keep-alive has failed and will no longer use this MQTT connection. */
+        _IotMqtt_DecrementConnectionReferences( pMqttConnection );
```

Comment:
What state is the connection left in if this happens?

Reply:
Connection will be closed by the _IotMqtt_CloseNetworkConnection call on line 850.